src/sinks/aws_cloudwatch_logs/config.rs

```diff
@@ -46,6 +46,17 @@ impl ClientBuilder for CloudwatchLogsClientBuilder {
     }
 }
 
+#[configurable_component]
+#[derive(Clone, Debug, Default)]
+/// Retention policy configuration for AWS Cloudwatch Log Group
+pub struct Retention {
+    #[configurable(derived)]
+    pub enabled: bool,
+
+    #[configurable(derived)]
+    pub days: i32,
```

Comment:
Add validation as this parameter supports only limited values set https://docs.aws.amazon.com/AmazonCloudWatchLogs/latest/APIReference/API_PutRetentionPolicy.html#API_PutRetentionPolicy_RequestSyntax